cpp/models/lct_secir/model.h

```diff
@@ -0,0 +1,102 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Lena Ploetzke
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef LCT_SECIR_MODEL_H
+#define LCT_SECIR_MODEL_H
+
+#include "lct_secir/parameters.h"
+#include "lct_secir/infection_state.h"
+#include "memilio/config.h"
+#include "memilio/utils/time_series.h"
+#include "memilio/math/eigen.h"
+#include <string>
+
+namespace mio
+{
+namespace lsecir
+{
+class Model
+{
+
+public:
+    /**
+     * @brief Constructor to create an LCT SECIR Model.
+     *
+     * @param[in] init Vector with initial values for all infection states inclusive subcompartments.
+     * @param[in, out] infectionState_init infectionState for the Model, specifies number of Subcompartments for each infection state.
+     * @param[in, out] parameters_init Specifies Parameters necessary for the Model. 
+     */
+    Model(Eigen::VectorXd init, InfectionState infectionState_init = InfectionState(),
+          Parameters&& parameters_init = Parameters());
+
+    /**
+     * @brief Checks constraints of the model inclusive check for model parameters.
+     */
+    bool check_constraints() const;
+
+    /**
+     * @brief Evaulates the right-hand-side f of the LCT dydt = f(y, t).
+     *
+     * The LCT-SECIR model is defined through ordinary differetial equations of the form dydt = f(y, t). 
+     * y is a vector containing number of individuals for each (sub-) compartment.
+     * This function evaluates the right-hand-side f of the ODE and can be used in an ODE solver.
+     * @param[in] y the current state of the model
+     * @param[in] t the current time
+     * @param[out] dydt a reference to the calculated output
+     */
+    void eval_right_hand_side(Eigen::Ref<const Eigen::VectorXd> y, ScalarType t,
+                              Eigen::Ref<Eigen::VectorXd> dydt) const;
+
+    /**
+     * @brief Calculates the population divided in states defined in InfectionStateBase out of a result with subcompartments.
```

Comment:
```suggestion
     * @brief Calculates the population divided in infection states defined in InfectionStateBase out of a result with subcompartments.
```
What do you mean by "out of a result with subcompartments"?

Reply:
Rewrote the brief, I think it should be clearer now.